tools/js_manipulation.py

```diff
@@ -133,22 +133,28 @@ def make_invoke(sig):
   # wasm won't implicitly convert undefined to 0 in this case.
   exceptional_ret = '\n    return 0n;' if legal_sig[0] == 'j' else ''
   body = '%s%s;' % (ret, make_dynCall(sig, args))
-  # Exceptions thrown from C++ exception will be integer numbers.
-  # longjmp will throw the number Infinity.
-  # Create a try-catch guard that rethrows the exception if anything else
-  # than a Number was thrown. To do that quickly and in a code size conserving
-  # manner, use the compact test "e !== e+0" to check if e was not a Number.
+  # Create a try-catch guard that rethrows the Emscripten EH exception.
+  if settings.EXCEPTION_STACK_TRACES:
+    # Exceptions thrown from C++ and longjmps will be an instance of
+    # EmscriptenEH.
+    rethrow = 'if (!(e instanceof EmscriptenEH)) throw e;'
```

Comment:
call this `maybe_rethrow`?

Reply:
3a3df48454908810bec6a69f0956f7181126f1f9